okteto/okteto
# fix: use okteto api to retrieve dev env endpoints

## Changes
- use okteto api to retrieve endpoints using `space` query
- use old logic for vanilla or fallbacks if endpoint is not available
- remove external endpoints retrieval when backend is used to retrieve endpoints

## How to test it
- clone okteto/movies
- cd to the repo
- deploy okteto/movies with an external resource (`okteto deploy`)
- check all endpoints are correct comparing them with endpoint in okteto UI when exec `okteto endpoints`
---
### With/without changes
- clone https://github.com/okteto/divert-with-istio-sample
- cd divert-with-istio-sample
- using my own cluster `okteto ctx use <my-context>`. You need to enable istio and an okteto feature flag in order to use virtual services so is easy to use my cluster already prepared.
- run `okteto deploy -f okteto-staging.yml`
- check endpoint in UI (broken because we need an ingress but enough to test the needed logic)
- run `okteto endpoints -f okteto-staging.yml` will show same endpoints than UI using this PR and no endpoints with previous implementation
